src/credentials/examples/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -367,6 +372,37 @@ AttestationVerificationResult DefaultDACVerifier::ValidateCertificateDeclaration
     return AttestationVerificationResult::kSuccess;
 }
 
+CHIP_ERROR DefaultDACVerifier::VerifyNodeOperationalCSRInformation(const ByteSpan & nocsrElementsBuffer,
+                                                                   const ByteSpan & attestationChallengeBuffer,
+                                                                   const ByteSpan & attestationSignatureBuffer,
+                                                                   const P256PublicKey & dacPublicKey, const ByteSpan & csrNonce)
+{
+    VerifyOrReturnError(!nocsrElementsBuffer.empty() && !attestationChallengeBuffer.empty() &&
+                            !attestationSignatureBuffer.empty() && !csrNonce.empty(),
+                        CHIP_ERROR_INVALID_ARGUMENT);
+
+    ByteSpan csrSpan;
+    ByteSpan csrNonceSpan;
+    ByteSpan vendorReserved1Span;
+    ByteSpan vendorReserved2Span;
+    ByteSpan vendorReserved3Span;
+    ReturnErrorOnFailure(DeconstructNOCSRElements(nocsrElementsBuffer, csrSpan, csrNonceSpan, vendorReserved1Span,
+                                                  vendorReserved2Span, vendorReserved3Span));
+
+    // Verify that Nonce matches with what we sent
+    VerifyOrReturnError(csrNonceSpan.data_equal(csrNonce), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Need a check that csrNonce length is == 32

Reply:
Addressed in f522e7e